src/bootstrap/check.rs

```diff
@@ -292,7 +292,7 @@ macro_rules! tool_check_step {
             const DEFAULT: bool = true;
 
             fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
-                run.path($path)
+                run.path($path) $(.path($alias))*
```

Comment:
This is not quite right -- we likely want to use paths(&[...]) here, which sets up a set of paths all describing the same thing, rather than different things on the same Step. We're not always consistent about that but it's the intended API usage.

This is basically intended to support the use case of test library/core library/std not testing alloc; these checks don't really support it anyway, but I'd prefer to use the right API. If it's a pain to make the macro play nice with that then no worries.

Reply:
Sure thing, done. It was pretty simple:
```diff
diff --git a/src/bootstrap/check.rs b/src/bootstrap/check.rs
index 786cca14863..536843d4ec2 100644
--- a/src/bootstrap/check.rs
+++ b/src/bootstrap/check.rs
@@ -292,7 +292,7 @@ impl Step for $name {
             const DEFAULT: bool = true;
 
             fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
-                run.path($path) $(.path($alias))*
+                run.paths(&[ $path, $($alias),* ])
             }
 
             fn make_run(run: RunConfig<'_>) {
```